### PR TITLE
Convert sample to kotlin

### PR DESCRIPTION
**Pull request by maxpeters**

### CODE REVIEW

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import com.jakewharton.rxbinding.view.RxView
+import net.grandcentrix.thirtyinch.TiActivity
+import net.grandcentrix.thirtyinch.logginginterceptor.LoggingInterceptor
+import net.grandcentrix.thirtyinch.sample.fragmentlifecycle.FragmentLifecycleActivity
+import net.grandcentrix.thirtyinch.sample.fragmentlifecycle.viewpager.LifecycleViewPagerActivity
+import rx.Observable
+
+class HelloWorldActivity : TiActivity<HelloWorldPresenter, HelloWorldView>(), HelloWorldView {
+
+    private lateinit var mButton: Button
```
Comment (StefMa) [inline sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt]: use `by lazy`

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            supportFragmentManager.beginTransaction()
+                    .replace(R.id.fragment_container, SampleFragment())
+                    .commit()
+        }
+
+        findViewById<View>(R.id.recreate).setOnClickListener { recreate() }
+    }
+
+    override fun onButtonClicked(): Observable<Void> {
+        return RxView.clicks(mButton)
```
Comment (StefMa) [inline sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt]: Could be used with the `= ` syntax:
```kotlin
override fun onButtonClicked() = RxView.clicks(mButton)
```

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            R.id.start_fragment_lifecycle_test -> {
+                startActivity(Intent(this, FragmentLifecycleActivity::class.java))
+                return true
+            }
+            R.id.start_viewpager_test -> {
+                startActivity(Intent(this, LifecycleViewPagerActivity::class.java))
+                return true
+            }
+        }
+        return false
```
Comment (StefMa) [inline sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt]: call super here please

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    override fun providePresenter(): HelloWorldPresenter {
```
Comment (StefMa) [inline sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt]: could be also used with the `= ` sytnax

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/SamplePresenter.kt [hunk 5]
```diff
@@ -13,29 +13,26 @@
...
+import java.util.concurrent.TimeUnit
+import net.grandcentrix.thirtyinch.TiPresenter
+import net.grandcentrix.thirtyinch.rx.RxTiPresenterSubscriptionHandler
+import net.grandcentrix.thirtyinch.rx.RxTiPresenterUtils
+import rx.Observable
 
+class SamplePresenter : TiPresenter<SampleView>() {
 
-public class SamplePresenter extends TiPresenter<SampleView> {
+    private val mSubscriptionHandler = RxTiPresenterSubscriptionHandler(this)
```
Comment (rehlma1989) [inline sample/src/main/java/net/grandcentrix/thirtyinch/sample/SamplePresenter.kt]: `mSubscriptionHandler -> subscriptionHandler`

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        return HelloWorldPresenter()
+    }
+
+    override fun showPresenterUpTime(uptime: Long?) {
```
Comment (StefMa) [inline sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldActivity.kt]: Is the Long really nullable?

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import rx.Observable
+import rx.schedulers.Schedulers
+import rx.subjects.BehaviorSubject
+import rx.subjects.PublishSubject
+import java.util.concurrent.TimeUnit
+
+class HelloWorldPresenter : TiPresenter<HelloWorldView>() {
+
+    private var mCounter = 0
+    private val mText = BehaviorSubject.create<String>()
```
Comment (StefMa) [inline sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldPresenter.kt]: don't use the `m` sytax in Kotlin

Comment (StefMa) [reply sample/src/main/java/net/grandcentrix/thirtyinch/sample/HelloWorldPresenter.kt]: Don't name it "text" but "textSubject" or so..

**Pull request merged**